monai/data/wsi_reader.py

```diff
@@ -103,6 +104,18 @@ def get_file_path(self, wsi) -> str:
         """Return the file path for the WSI object"""
         raise NotImplementedError(f"Subclass {self.__class__.__name__} must implement this method.")
 
+    @abstractmethod
+    def get_mpp(self, wsi, level: Optional[int] = None) -> Tuple[float, float]:
+        """
+        Returns the micro-per-pixel resolution of the whole slide image at a given level.
+
+        Args:
+            wsi: a whole slide image object loaded from a file
+            level: the level number where the size is calculated
+
+        """
+        raise NotImplementedError(f"Subclass {self.__class__.__name__} must implement this method.")
+
     @abstractmethod
     def get_patch(
         self, wsi, location: Tuple[int, int], size: Tuple[int, int], level: int, dtype: DtypeLike, mode: str
```

Comment:
Maybe this `level` args can also be optional?

Reply:
Thanks for your comment but no need to do that here. It is a lower level method that user shouldn't use. I rename them and prepend `_` to denote that they are private methods.